sled-agent/types/src/rack_init.rs

```diff
@@ -298,19 +299,20 @@ impl RackInitializeRequest {
     pub fn from_toml_with_fallback(
         data: &str,
     ) -> Result<RackInitializeRequest> {
-        let v2_err = match toml::from_str::<RackInitializeRequest>(&data) {
+        match toml::from_str::<RackInitializeRequest>(&data) {
             Ok(req) => return Ok(req),
```

Comment:
Tiny nit - with the rework here I think we could drop this return?

```suggestion
            Ok(req) => Ok(req),
```